plutus-core/cost-model/create-cost-model/CreateBuiltinCostModel.hs

```
@@ -235,241 +228,217 @@ createBuiltinCostModel bmfile rfile = do
+
+-- | A two-variable costing function which is constant on one region of the
+-- plane and something else elsewhere.
+readTwoVariableFunConstOr :: MonadR m => (SomeSEXP s -> m ModelTwoArguments) -> SomeSEXP s -> m ModelConstantOrTwoArguments
+readTwoVariableFunConstOr reader e = do
+  constantPart <- getExtraParam "default" e
+  nonConstantPart <- reader e
+  pure $ ModelConstantOrTwoArguments constantPart  nonConstantPart
+
+
+-- | Functions to read CPU costing functions from R.  There are some costing
+-- function types which are currently only used for memory models (which are
+-- constructed directly, not via R), and those won't be handled here.
+
+loadOneVariableCostingFunction :: MonadR m => SomeSEXP (Region m) -> m ModelOneArgument
+loadOneVariableCostingFunction e = do
+  ty <- getType e
+  case ty of
+    "constant_cost" -> ModelOneArgumentConstantCost <$> getConstant e
+    "linear_cost"   -> ModelOneArgumentLinearCost <$> readOneVariableLinearFunction "x_mem" e
+    "linear_in_x"   -> ModelOneArgumentLinearCost <$> readOneVariableLinearFunction "x_mem" e  -- FIXME: duplicate
+    _               -> error $ "Unknown one-variable model type: " ++ ty
+
+loadTwoVariableCostingFunction :: MonadR m => SomeSEXP (Region m) -> m ModelTwoArguments
+loadTwoVariableCostingFunction e = do
+  ty <- getType e
+  case ty of
+    "constant_cost"        -> ModelTwoArgumentsConstantCost       <$> getConstant e
+    "linear_in_x"          -> ModelTwoArgumentsLinearInX          <$> readOneVariableLinearFunction "x_mem" e
+    "linear_in_y"          -> ModelTwoArgumentsLinearInY          <$> readOneVariableLinearFunction "y_mem" e
+    "linear_in_x_and_y"    -> ModelTwoArgumentsLinearInXAndY      <$> readTwoVariableLinearFunction "x_mem" "y_mem" e
+    "added_sizes"          -> ModelTwoArgumentsAddedSizes         <$> readOneVariableLinearFunction "I(x_mem + y_mem)" e
+    "subtracted_sizes"     -> ModelTwoArgumentsSubtractedSizes    <$> error "subtracted sizes"
```

Why do we need this line given the catch-all clause?

---

That was left over from when I was trying to get it to work.  I've removed it now.  We don't use this function (or the other one later) for CPU usage, so we don't need a case for them.